### ISSUE TITLE
can't repair a malfunctioned EVA suit

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#689]
**Describe the bug**
- If an EVA suit has a malfunction such as a "Class A fire", the malfunction gets reported in a log statement. However, the malfunctioned EVA suit can't be repaired.

**Expected and unexpected behavior**
- Equipment repair should be captured and dealt with by `RepairInsideMalfunction` or a brand new class.  - - A vehicle is also considered an equipment. If a vehicle is outside, the malfunctioned vehicle could be repaired by `RepairEVAMalfunction`. 

**Java classes involved**
- Equipment repair is not done properly by `RepairInsideMalfunction` or `RepairEVAMalfunction`. 

**Screenshots**
![image](https://user-images.githubusercontent.com/1168584/190837955-dff55adc-cbf0-4276-af5c-7a240f34c80b.png)
Note that this broken EVA suit 042 is NOT captured in the Swing UI in Settlement's Malfunction tab since it only display building malfunctions and not equipment's.
![image](https://user-images.githubusercontent.com/1168584/190838196-65873571-c76c-4ff1-8793-d92946b75df4.png)
Also, the Monitor Tool failed to capture EVA suit malfunction in 'Tian Cheng'
![image](https://user-images.githubusercontent.com/1168584/190838221-547009cb-8a0b-4f9f-b6b1-34653093384a.png)

**Stacktrace**
The log statement below was printed excessively, indicating that a malfunctioned was somehow detected by `RepairInsideMalfunctionMeta` but failed to be found in RepairInsideMalfunction`'s constructor. 
> "Could not find a malfunction to work on in my vicinity."

**Specifications  (please complete the following information):**
 - pre-3.4.0 build 7327

**Additional context**
- If following the separation of concern principle, we should create a separate task and meta task to deal with standalone equipment repair.
- In that case, `RepairInsideMalfunction` and `RepairEVAMalfunction` will be used exclusively for building structural repair. 
- Similarly, we will need to consider what's best for vehicle repair and robot repair in the long run.
 
**Related Issues**
- https://github.com/mars-sim/mars-sim/issues/597
- https://github.com/mars-sim/mars-sim/issues/377

**Comment by mokun** [comment]
@bevans2000 

I'll commit a temporary solution, but in the long run, do you agree we should refine and break out equipment repair onto a separate task ?

Can you artificially induce a malfunction on an EVA suit in text console to see if it works ?

**Comment by bevans2000** [comment]
Yes equipment repair should be a separate task

**Comment by mokun** [comment]
A question on `RepairEVAMalfunction` class:

```
	public RepairEVAMalfunction(Person person) {
		super(NAME, person, true, 25, SkillType.MECHANICS);

		if (!person.isNominallyFit()) {
			checkLocation();
        	return;
		}

		if (person.isInSettlement()) {
			containerUnit = person.getSettlement();
		}
		else if (person.isInSettlementVicinity()) {
			containerUnit = person.getNearbySettlement();
		}
		else if (person.isInVehicle()){
			containerUnit = person.getVehicle();
		}
		else if (person.getMission() != null) {
			// if the person is out on mission and not within a settlement vicinity
			Mission mission = person.getMission();
			containerUnit = ((VehicleMission)mission).getVehicle();
		}

		if (containerUnit != null) {
			// Get the malfunctioning entity.
			for (Malfunctionable next : MalfunctionFactory.getLocalMalfunctionables(person)) {
				Malfunction potential = next.getMalfunctionManager().getMostSeriousMalfunctionInNeed(MalfunctionRepairWork.EVA);
				if (potential != null) {
					entity = next;
					malfunction = potential;
					break; // Stop searching
				}
			}
		}

		// Start if found
		if (entity != null) {
			setDescription(Msg.getString("Task.description.repairEVAMalfunction.detail", malfunction.getName(),
					entity.getNickName())); // $NON-NLS-1$

			// Determine location for repairing malfunction.
			setOutsideLocation((LocalBoundedObject) entity);

			// Can fail to get a path and Task will be Done
			if (!isDone()) {
	            setPhase(WALK_TO_OUTSIDE_SITE);

				RepairHelper.startRepair(malfunction, person, MalfunctionRepairWork.EVA, entity);

				// Initialize phase
				addPhase(REPAIRING);
			}
		}
		else {
			checkLocation();
		}
	}
```

I can see it calls out `addPhase(REPAIRING)` and even `setPhase(**REPAIRING**)`.

But there's no `setPhase(REPAIRING)`.

I suppose it will automatically call up `setPhase(REPAIRING)` when it's the right time ?!

**Comment by bevans2000** [comment]
That's an interesting find. There definitely needs to be an active phase for all Tasks. I can have a look by debugging it

**Comment by bevans2000** [comment]
@mokun I've just changed the 'create malfunction' Console Command so it allows a contained Equipment to be the target of the Malfunction. To use it connect to a Unit that holds Equipment and run the command; then any EVA Suit contained can be made to fail. It's particularly useful to run against a Person on EVA as their suit can be made to fail.
I've chosen this approach because list all Equipment from a single list is too much.

**Comment by mokun** [comment]
Awesome !

> It's particularly useful to run against a Person on EVA as their suit can be made to fail.

Now, I have a brand new question: what do we want to see if the suit fails during EVA ?

**Comment by bevans2000** [comment]
Person should end it and attempt to return. Then you have to handle them jumping the queue at the airlock?

**Comment by mokun** [comment]
> jumping the queue at the airlock?

do you mean skipping the usual steps of ingressing ?

**Comment by bevans2000** [comment]
Fixed in #778
